### PR TITLE
Add validations for booking updation.

### PR DESCRIPTION
**Pull request by aditi-webonise**
Add mappings among entities.
Format the view.

### CODE REVIEW

File: src/main/java/com/webonise/rbs/controller/BookingController.java [hunk 1]
```diff
@@ -30,8 +30,10 @@ public String showBookings(Model model) {
 
     @PostMapping
     public String saveBooking(@ModelAttribute("booking") Booking booking, final RedirectAttributes redirectAttributes) {
-        if(bookingService.bookRoom(booking)!= null) {
-            redirectAttributes.addFlashAttribute("saveBooking", "success");
+        if (bookingService.checkBookingAvailability(booking)) {
+            if (bookingService.bookRoom(booking)!=null) {
+                redirectAttributes.addFlashAttribute("saveBooking", "success");
```
Comment (akshaym-webonise) [inline src/main/java/com/webonise/rbs/controller/BookingController.java]: what if **bookingService.bookRoom(booking)** returns null.

File: src/main/java/com/webonise/rbs/constants/RedirectStatus.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package com.webonise.rbs.constants;
+
+public enum RedirectStatus {
+
+    success("success"),
```
Comment (himani-webonise) [inline src/main/java/com/webonise/rbs/constants/RedirectStatus.java]: Make it in uppercase, Enum member are constant please follow the convention of constant.

File: src/main/java/com/webonise/rbs/controller/BookingController.java [hunk 3]
```diff
@@ -21,49 +24,59 @@
...
+
+    @Autowired
+    private EventService eventService;
+
     @GetMapping
-    public String showBookings(Model model) {
-        model.addAttribute("booking", new Booking());
-        model.addAttribute("allBookings", bookingService.getAllBookings());
+    public String showBookings (Model model) {
+        model.addAttribute("booking" , new Booking());
```
Comment (himani-webonise) [inline src/main/java/com/webonise/rbs/controller/BookingController.java]: Please follow the standard of code format, remove space before comma(,)
A comma (,) always stay attached to the token that precedes whether it's programming language or language of communication.

File: src/main/java/com/webonise/rbs/controller/BookingController.java [hunk 4]
```diff
@@ -21,49 +24,59 @@
...
+        model.addAttribute("allBookings" , bookingService.getAllBookings());
```
Comment (himani-webonise) [inline src/main/java/com/webonise/rbs/controller/BookingController.java]: The above comment is applied everywhere.

File: src/main/java/com/webonise/rbs/service/RoleServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+import com.webonise.rbs.entity.Role;
+import com.webonise.rbs.repository.RoleRepository;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional
+public class RoleServiceImpl implements RoleService{
```
Comment (himani-webonise) [inline src/main/java/com/webonise/rbs/service/RoleServiceImpl.java]: Give space between RoleService and {

File: src/main/java/com/webonise/rbs/service/RoleServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+    @Autowired
+    private RoleRepository roleRepository;
+
+    @Override
+    public List<Role> getAllRoles() {
+        return roleRepository.findAll();
+    }
+}
```
Comment (himani-webonise) [inline src/main/java/com/webonise/rbs/service/RoleServiceImpl.java]: Please add new line at the EOF

**Comment by himani-webonise** [comment]
PR looks good to me

**Review by himani-webonise — APPROVED** [review]

**Pull request merged**